test/framework/clusterstatevalidations/vsphere.go

```diff
@@ -0,0 +1,46 @@
+package clusterstatevalidations
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	clusterf "github.com/aws/eks-anywhere/test/framework/cluster"
+)
+
+// ValidateCSI checks that the CSI is enabled/disabled based on the settings in the cluster.Spec by
+// by checking if CSI related objects components exist in the cluster or not.
+func ValidateCSI(ctx context.Context, vc clusterf.StateConfig) error {
+	clusterClient := vc.ClusterClient
+
+	yaml := vc.ClusterSpec.Config.VSphereDatacenter
+	yamlCSI := yaml.Spec.DisableCSI
+
+	deployment := &v1.Deployment{}
+	deployKey := types.NamespacedName{Namespace: "kube-system", Name: "vsphere-csi-controller"}
+	deployErr := clusterClient.Get(ctx, deployKey, deployment)
+	deployRes := handleCSIError(deployErr, yamlCSI)
+	if deployRes != nil {
+		return deployRes
+	}
+
+	ds := &v1.DaemonSet{}
+	dsKey := types.NamespacedName{Namespace: "kube-system", Name: "vsphere-csi-node"}
+	dsErr := clusterClient.Get(ctx, dsKey, ds)
+	dsRes := handleCSIError(dsErr, yamlCSI)
+	if dsRes != nil {
+		return dsRes
+	}
+
+	return nil
+}
+
+func handleCSIError(err error, disabled bool) error {
+	if (disabled && err == nil) || (!disabled && err != nil) {
```

Comment:
there could be an error for other reasons that not found
I feel like you could also handle the `NamespacedName` and client.Get here and simplify the code in the caller.

Reply:
Should we handle this as a fix in another PR as well? This PR is just moving the validations.